src/components/TextField/TextField.scss

```diff
@@ -388,6 +388,24 @@ $stacking-order: (
     }
   }
 
+  // This is a violation of our component model, but necessary
+  // due to Connected.scss overriding these values.
+  // stylelint-disable declaration-no-important
+
+  .ConnectedLeft {
+    &::after {
+      border-top-left-radius: 0 !important;
+      border-bottom-left-radius: 0 !important;
+    }
+  }
+
+  .ConnectedRight {
+    &::after {
+      border-top-right-radius: 0 !important;
+      border-bottom-right-radius: 0 !important;
+    }
+  }
```

Comment:
@dleroux is this what you were aiming for?

Reply:
Sort of. I just tested and if you do the following you will not need the important and it will generate this className which make sense:

`.Polaris-TextField__Backdrop--connectedLeft::after`

```
.Backdrop {
    @include focus-ring($border-width: border-width());
    background-color: var(--p-surface);
    position: absolute;

    &::after {
      opacity: 1;
    }
  }

  .Backdrop-connectedLeft::after {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }

  .Backdrop-connectedRight::after {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
```